idseq/uploader.py

```diff
@@ -56,21 +56,22 @@ def split_file(self, max_part_size, prefix):
 
         partial_files = []
         suffix_iter = product(ascii_lowercase, repeat=2)
-        try:
-            buffer = memoryview(bytearray(min(BUFFER_SIZE, max_part_size)))
-            with buffer, open(self.path, 'rb') as fread:
-                for suf in suffix_iter:
+        buffer = memoryview(bytearray(min(BUFFER_SIZE, max_part_size)))
+        with buffer, open(self.path, 'rb') as fread:
+            for suf in suffix_iter:
+                remaining = max_part_size
+                bytes_read = fread.readinto(buffer)
+                if bytes_read:
                     partial_files.append("{}{}".format(prefix, ''.join(suf)))
-                    remaining = max_part_size
-                    bytes_read = fread.readinto(buffer)
-                    if bytes_read:
-                        with open(partial_files[-1], 'wb') as fwrite:
-                            while bytes_read:
-                                fwrite.write(buffer[:bytes_read])
-                                remaining -= bytes_read
-                                bytes_read = fread.readinto(buffer[:min(remaining, len(buffer))])
-            return partial_files
-        except StopIteration:
```

Comment:
Can you add a comment on the reasons why the iteration might break? Or can we make the message more specific?
Is it just the suffixes?
How does this relate to the presigned_url?

Reply:
I mean, what is the first bottleneck? The suffixes or the size of the url?

Reply:
The old version was using `next(suffix_iter)` and manually handling the `StopIteration`.
I replaced that part with a `for` loop, but I left the `partial_files.append(...)` outside the `if bytes_read` condition that I introduced to control the read buffer. 
The code also wasn't breaking the loop once the original file was fully read, which wasn't an issue for file split, because the `if bytes_read:` was guarding that condition. However, all the zero-byte parts ended up being added to that array.